source/NetworkUtils.cpp

```diff
@@ -270,4 +270,169 @@ void NetworkUtils::broadcastStartSession(ENetHost* serverHost) {
+		outBuffer.writeBool8((*player->axisLock)[2]);
+
+		outBuffer.writeBool8(*player->accelerationEnabled);
+		outBuffer.writeFloat32(*player->movementAcceleration);
+		outBuffer.writeFloat32(*player->movementDeceleration);
+
+		outBuffer.writeFloat32(*player->sprintMultiplier);
+
+		outBuffer.writeFloat32(*player->jumpVelocity);
+		outBuffer.writeFloat32(*player->jumpInterval);
+		outBuffer.writeBool8(*player->jumpTouch);
+
+		outBuffer.writeFloat32(*player->height);
+		outBuffer.writeFloat32(*player->crouchHeight);
+
+		outBuffer.writeBool8(*player->headBobEnabled);
+		outBuffer.writeFloat32(*player->headBobAmplitude);
+		outBuffer.writeFloat32(*player->headBobFrequency);
+
+		outBuffer.writeVector3(*player->respawnPos);
+		outBuffer.writeBool8(true);
+
+		outBuffer.writeFloat32(*player->movementRestrictionX);
+		outBuffer.writeFloat32(*player->movementRestrictionZ);
+		outBuffer.writeBool8(*player->restrictedMovementEnabled);
+
+		outBuffer.writeBool8(*player->counterStrafing);
+	}
+	else {
+		int index = (*player->selectedClient).compare("Client 1") ? index = 0 : index = 1;
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].moveRate);
+		outBuffer.writeVector2((*player->clientPlayerConfigs)[index].moveScale);
+
+		outBuffer.writeBool8(((*player->clientPlayerConfigs)[index].axisLock)[0]);
+		outBuffer.writeBool8(((*player->clientPlayerConfigs)[index].axisLock)[1]);
+		outBuffer.writeBool8(((*player->clientPlayerConfigs)[index].axisLock)[2]);
+
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].accelerationEnabled);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementAcceleration);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementDeceleration);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].sprintMultiplier);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].jumpVelocity);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].jumpInterval);
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].jumpTouch);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].height);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].crouchHeight);
+
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].headBobEnabled);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].headBobAmplitude);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].headBobFrequency);
+
+		outBuffer.writeVector3((*player->clientPlayerConfigs)[index].respawnPos);
+		outBuffer.writeBool8(true);
+
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementRestrictionX);
+		outBuffer.writeFloat32((*player->clientPlayerConfigs)[index].movementRestrictionZ);
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].restrictedMovementEnabled);
+
+		outBuffer.writeBool8((*player->clientPlayerConfigs)[index].counterStrafing);
+
+		*player->readFromFile = false;
```

Comment:
Why does writing this change `readFromFile`? Perhaps I'm misunderstanding this as a control flag when it's actually a semaphore? I'd clarify this where its declared and here w/ a comment stating what it is supposed to do

Reply:
As I did not use a checkbox in the GUI for the "Reading from Configs" functionality, this currently works as a semaphore. Which is set by the button, and after we are done reading, we set it to false.